### ISSUE TITLE
Typing activities blow up in live bot

### ISSUE THREAD

**Issue by brandonh-msft** [Microsoft/botbuilder-dotnet#460]
```csharp
                        await turn.SendActivities(new[]{
                            MessageFactory.Text(@"Stopping server..."),
                            Activity.CreateTypingActivity()});

```

Works great in emulator, blows up in Azure with a nullref exception down in Botbuilder core.

also tried

```csharp
                        var typing = turn.Activity.CreateReply();
                        typing.Type = ActivityTypes.Typing;
                        await turn.SendActivity(typing);
```

w/ the same result.

**Comment by drub0y** [comment]
Would you be so kind as to edit and include the full stack trace?

![please](https://media.giphy.com/media/aZ4sQUpybai5y/giphy.gif)

**Comment by cleemullins** [comment]
That's not good. Working on a Repro now. Can you include the stack trace? That would make this much quicker to turn-around.

**Comment by Stevenic** [comment]
@brandonh-msft  what channel are you using out of curiosity?

**Comment by cleemullins** [comment]
As a quick workaround, here's Jason's comment on this in the [v3 SDK](https://stackoverflow.com/questions/49908140/bot-framework-cant-get-activitytypes-typing-to-work/49910627#49910627) earlier today.

It's probably the same problem, with the same fix.

**Comment by cleemullins** [comment]
@JasonSowers  has confirmed this is working properly. 
"the second code snippet is working in emulator, webchat, and facebook.  However in webchat it is not actually displaying the typing indicator"

**Comment by JasonSowers** [comment]
@brandonh-msft using the code below I was able to send a typing activity in the emulator, facebook, and webchat.  It did not blow up at all the only thing that was an issue for me was that in webchat the typing indicator was not actually displayed, but the code ran just fine.  Can you please provide more detail and context?

```cs
using System.Threading.Tasks;
using Microsoft.Bot;
using Microsoft.Bot.Builder;
using Microsoft.Bot.Builder.Core.Extensions;
using Microsoft.Bot.Schema;

namespace AspNetCore_EchoBot_With_State
{
    public class EchoBot : IBot
    {
        /// <summary>
        /// Every Conversation turn for our EchoBot will call this method. In here
        /// the bot checks the Activty type to verify it's a message, bumps the 
        /// turn conversation 'Turn' count, and then echoes the users typing
        /// back to them. 
        /// </summary>
        /// <param name="context">Turn scoped context containing all the data needed
        /// for processing this conversation turn. </param>        
        public async Task OnTurn(ITurnContext context)
        {
            // This bot is only handling Messages
            if (context.Activity.Type == ActivityTypes.Message)
            {
                // Get the conversation state from the turn context
                //var state = context.GetConversationState<EchoState>();
                
                var typing = context.Activity.CreateReply();
                typing.Type = ActivityTypes.Typing;
                await context.SendActivity(typing);
                // Bump the turn count. 
                //state.TurnCount++;

                // Echo back to the user whatever they typed.
                //await context.SendActivity($"Turn {state.TurnCount}: You sent '{context.Activity.Text}'");
            }
        }
    }    
}
```

**Comment by JasonSowers** [comment]
I think Steve is correct.  It blew up in Teams stack trace below:

```
   at Microsoft.Bot.Builder.TurnContext.<>c__DisplayClass22_0.<<SendActivities>g__ActuallySendStuff|1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Bot.Builder.TurnContext.<SendActivitiesInternal>d__26.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Bot.Builder.TurnContext.<SendActivities>d__22.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Bot.Builder.TurnContext.<SendActivity>d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at AspNetCore_EchoBot_With_State.EchoBot.<OnTurn>d__0.MoveNext() in C:\source\gitHub\botbuilder-dotnet\AspNetCore-EchoBot-With-State\EchoBot.cs:line 32
```

**Comment by JasonSowers** [comment]
Yup @Stevenic  is right the following code worked:

```cs
                    var typing = context.Activity.CreateReply();
                    typing.Type = ActivityTypes.Typing;
                    typing.ReplyToId = context.Activity.Recipient.Id;
                    await context.SendActivity(typing);
```

**Comment by cleemullins** [comment]
@JasonSowers will update CreateReply tomorrow to make sure this is always filled out. Note that it *only* fails with Teams. 

Note: I think this thread highlights that Bug Reports really need to be more detailed. I'll talk with the team tomorrow, and make sure we have a Bug Template.

**Comment by JasonSowers** [comment]
That code is no longer working after about 5 successfully sent typing activities, it blows up everytime now.

**Comment by brandonh-msft** [comment]
My example occurred in Skype.

**Comment by cleemullins** [comment]
Fixed in #480
